gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,203 @@
+   * The message will be ack'ed upon successful return or nack'ed if exception is thrown.
+   */
+  interface PullCallback {
+    void process(Message message) throws Exception;
+  }
+
+  Topic create(TopicInfo topic);
+
+  Future<Topic> createAsync(TopicInfo topic);
+
+  // null if not found
+  Topic getTopic(String topic);
+
+  Future<Topic> getTopicAsync(String topic);
+
+  // false if not found
+  boolean deleteTopic(String topic);
+
+  Future<Boolean> deleteTopicAsync(String topic);
+
+  Page<Topic> listTopics(ListOption... options);
+
+  Future<AsyncPage<Topic>> listTopicsAsync(ListOption... options);
+
+  String publish(String topic, Message message);
+
+  Future<String> publishAsync(String topic, Message message);
+
+  List<String> publish(String topic, Message message, Message... messages);
+
+  Future<List<String>> publishAsync(String topic, Message message, Message... messages);
+
+  List<String> publish(String topic, Iterable<Message> messages);
+
+  Future<List<String>> publishAsync(String topic, Iterable<Message> messages);
+
+  Subscription create(SubscriptionInfo subscription);
+
+  Future<Subscription> createAsync(SubscriptionInfo subscription);
+
+  // null if not found
+  Subscription getSubscription(String subscription);
+
+  Future<Subscription> getSubscriptionAsync(String subscription);
+
+  void replacePushConfig(String subscription, PushConfig pushConfig);
+
+  Future<Void> replacePushConfigAsync(String subscription, PushConfig pushConfig);
+
+  // false if not found
+  boolean deleteSubscription(String subscription);
+
+  Future<Boolean> deleteSubscriptionAsync(String subscription);
+
+  Page<Subscription> listSubscriptions(ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(ListOption... options);
+
+  Page<Subscription> listSubscriptions(String topic, ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(String topic, ListOption... options);
+
+  List<ReceivedMessage> pull(String subscription, PullOption... options);
+
+  Future<List<ReceivedMessage>> pullAsync(String subscription, PullOption... options);
+
+  void pull(String subscription, PullCallback callback, PullOption... options);
+
+  void pullAsync(String subscription, PullCallback callback, PullOption... options);
+
+  void ack(String subscription, String ackId, String... ackIds);
+
+  Future<Void> ackAsync(String subscription, String ackId, String... ackIds);
+
+  void ack(String subscription, Iterable<String> ackIds);
+
+  Future<Void> ackAsync(String subscription, Iterable<String> ackIds);
+
+  void nack(String subscription, String ackId, String... ackIds);
+
+  Future<Void> nackAsync(String subscription, String ackId, String... ackIds);
+
+  void nack(String subscription, Iterable<String> ackIds);
+
+  Future<Void> nackAsync(String subscription, Iterable<String> ackIds);
+
+  void modifyAckDeadline(String subscription, int deadline, TimeUnit unit, String ackId,
```

Comment:
If I understood it correctly then in `Iterator<ReceivedMessage> pull(String subscription, PullOption... options);` we automatically renew the deadline for a message until it's read with `.next()`. In that case having this method is still necessary (for the user to implement his own renewal if he needs to).